### PR TITLE
 📖 Edit and rework checks.md (via checks.yaml and main.go)

### PR DESCRIPTION
**Pull request by olivekl**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Docs update

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:
Edit language for clarity; reorganize content; add formatting for easier reading; highlight risk levels and add short summary of reasons for check's risk level; add links and context where needed; fix typos and small errors.  Changes made to checks.yml and main.go to generate checks.md.

### CODE REVIEW

**Review by azeemshaikh38 — APPROVED** [review]

**Comment by azeemshaikh38** [comment]
LGTM. @laurentsimon @naveensrinivasan @david-a-wheeler in case anyone is interested in giving this a look before we submit.

**Review by naveensrinivasan — APPROVED** [review]
Looks good to me. Thanks!

File: docs/checks.md [hunk 1]
```diff
@@ -2,159 +2,479 @@
...
+    package via a feed that was not expected for that package.
+
+However, pinning dependencies can inhibit software updates, either because of a
+security vulnerability or because the pinned version is compromised. Mitigate
+this risk by:
+
+  - [having applications and _not_ libraries pin to specific versions](https://jbeckwith.com/2019/12/18/package-lock/);
+  - using automated tools to notify applications when their dependencies are
+    outdated;
+  - using applications that _do_ pin dependencies update quickly.
```
Comment (olivekl) [inline docs/checks.md:394]: Just wanted to highlight this line to be double-checked for accuracy and clarity.

Comment (olivekl) [reply docs/checks.md]: Thanks. Line 394 is updated.

File: docs/checks/internal/checks.yaml [hunk 2]
```diff
@@ -123,44 +124,49 @@ checks:
...
-      (such as child pornography), it may be impractical to hide the commit
-      and in such cases the history may need to be rewritten.
-
-      This check determines if the default and release branches are
-      protected with GitHub's branch protection settings.
-      The check only works when the token has [Admin
-      access](https://github.community/t/enable-branch-protection-get-api-without-admin/14197)
-      to the repository. This check determines if the default and release branches are
-      protected.
+    description: |
```
Comment (laurentsimon) [inline docs/checks/internal/checks.yaml:127]: maybe add a note about https://github.com/ossf/scorecard/issues/1106#issuecomment-936591864 somewhere? Just note that these settings in the link are not accessible without admin token. Or in the main readme.. or both. I don't know which is best.

**Pull request merged**